### DISCUSSION TITLE
Move rich text implimentation as default text renderer

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [godotengine/godot-proposals#7595] [GUI]
### Describe the project you are working on

Spaceship Game

### Describe the problem or limitation you are having in your project

I would like to use ui elements with rich text effect depending on the ui state

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Move rich text to be the default text renderer and add custom rich text effects to control and label3d.
Each control state would have the ability to add text overloads.
For example button could have different rich text based on normal, pressed, hovered, focused, disabled

This allows us to have way more control over text in godot

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

see last

### If this enhancement will not be used often, can it be worked around with a few lines of script?

This can be worked around in cases like buttons which you can just add a child but doesn't if it's something like tab container which is a composite control

### Is there a reason why this should be core and not an add-on in the asset library?

Rich text is already part of core but it only works as a separate label node.

**Comment by Zireael07** [comment]
Text in Godot can already be performance heavy, and RTL is heavier than Label
